pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,159 @@ type ServiceExportList struct {
+	// to a control plane / data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Accepted"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "NoService"
+	// * "InvalidServiceType"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionAccepted ServiceExportConditionType = "Accepted"
+
+	// ServiceExportReasonAccepted is used with the "Accepted" condition when the
+	// condition is True.
+	ServiceExportReasonAccepted ServiceExportConditionReason = "Accepted"
+
+	// ServiceExportReasonNoService is used with the "Accepted" condition when
+	// the associated Service does not exist.
+	ServiceExportReasonNoService ServiceExportConditionReason = "NoService"
+
+	// ServiceExportReasonInvalidServiceType is used with the "Accepted"
+	// condition when the associated Service has an invalid type
+	// (per the KEP at least the ExternalName type).
+	ServiceExportReasonInvalidServiceType ServiceExportConditionReason = "InvalidServiceType"
+)
+
+const (
+	// ServiceExportConditionExported is true when the service is exported to some
+	// control plane or data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Exported"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Pending"
+	// * "Failed"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionExported ServiceExportConditionType = "Exported"
+
+	// ServiceExportReasonExported is used with the "Exported" condition
+	// when the condition is True.
+	ServiceExportReasonExported ServiceExportConditionReason = "Exported"
+
+	// ServiceExportReasonPending is used with the "Exported" condition
+	// when the service is in the process of being exported.
+	ServiceExportReasonPending ServiceExportConditionReason = "Pending"
+
+	// ServiceExportReasonFailed is used with the "Exported" condition
+	// when the service failed to be exported with the message providing
+	// the specific reason.
+	ServiceExportReasonFailed ServiceExportConditionReason = "Failed"
+)
+
+const (
+	// ServiceExportConditionConflicted indicates that some property of an
+	// exported service has conflicting values across the constituent
+	// ServiceExports. This condition must be at least raised on the
+	// conflicting ServiceExport and is recommended to be raised on all on
+	// all the constituent ServiceExports if feasible.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "PortConflict"
+	// * "TypeConflict"
+	// * "SessionAffinityConflict"
+	// * "SessionAffinityConfigConflict"
+	// * "AnnotationsConflict"
+	// * "LabelsConflict"
```

Comment:
hmm, it's not easy to extend, later we may support other field conflicts.

How about just defining the "ConflictFound" and putting the details in the message?

Reply:
I would prefer to have specific reasons IMO, you are free to use your own reasons in case your use case is not supported as stated in this file.

Reply:
The conformance test generally present targeted conflict with one property conflicting so it should be relatively fine to make this into the conformance test. If there are test with multiple conflict we would probably not test the exact reason I would say. Also if you have particular situation that are specific to your implementation it would probably not be in the conformance test to begin with.